addons/arsenal/functions/fnc_handleStats.sqf

```diff
@@ -43,222 +44,217 @@ private _hideUnusedFnc = {
-            (0.5 - WIDTH_TOTAL / 2) + WIDTH_GAP,
-            safezoneY + 1.8 * GRID_H,
-            47 * GRID_W,
-            _height * GRID_H
-        ];
-        _statsBoxCtrl ctrlCommit 0;
-        private _ctrl = _display displayCtrl IDC_actionsBox;
-        private _pos = ctrlPosition _ctrl;
-        _pos set [1, safezoneY + (_height + 3.6) * GRID_H];
-        _ctrl ctrlSetPosition _pos;
-        _ctrl ctrlCommit 0;
+    // Move action display
+    private _ctrl = _display displayCtrl IDC_actionsBox;
+    private _pos = ctrlPosition _ctrl;
+    _pos set [1, safezoneY + 1.8 * GRID_H];
+    _ctrl ctrlSetPosition _pos;
+    _ctrl ctrlCommit 0;
+
+    {
+        _x ctrlSetFade 1;
+        _x ctrlCommit 0;
+    } forEach [
+        _statsPreviousPageCtrl,
+        _statsNextPageCtrl,
+        _statsCurrentPageCtrl
+    ];
+};
 
+// If nothing is chosen, hide stats
+if (isNull _itemCfg) exitWith {
+    call _fnc_hideEverything
+};
 
-        GVAR(statsInfo) = [_leftPanel, _statsIndex, _control, _curSel, _itemCfg];
+private _fnc_handleStats = {
+    params ["_statsIndex", "_leftPanel"];
 
-        // Toggle page buttons
-        _statsPreviousPageCtrl ctrlEnable (_currentPage > 0);
-        _statsNextPageCtrl ctrlEnable (_currentPage + 1 < count _statsArray);
-        _statsCurrentPageCtrl ctrlSetText ([LLSTRING(page), str (_currentPage + 1)] joinString " ");
+    private _statsPanel = [GVAR(statsListRightPanel), GVAR(statsListLeftPanel)] select _leftPanel;
 
-        {
-            _x ctrlSetFade 0;
-            _x ctrlCommit 0;
-        } forEach [
-            _statsPreviousPageCtrl,
-            _statsNextPageCtrl,
-            _statsCurrentPageCtrl
-        ];
+    // Get all viable stats for this tab
+    private _statsTab = _statsPanel select _statsIndex select {
+        _x params ["", "_configEntry", "", "", "_statements"];
+        _statements params ["", "", ["_condition", {true}, [{}]]];
+
+        ([_configEntry, _itemCfg] call _condition)
     };
 
-    // Check if in left or right panel
-    if (ctrlIDC _control == IDC_leftTabContent) then {
-        // Faces, voices and insigna do not have stats
-        if ([IDC_buttonFace, IDC_buttonVoice, IDC_buttonInsignia] find GVAR(currentLeftPanel) > -1) then {
-            [[1, 2, 3, 4, 5]] call _hideUnusedFnc;
-            _statsBoxCtrl ctrlSetPosition [
-                (0.5 - WIDTH_TOTAL / 2) + WIDTH_GAP,
-                safezoneY + 1.8 * GRID_H,
-                47 * GRID_W,
-                11 * GRID_H
-            ];
-            _statsBoxCtrl ctrlCommit 0;
-
-            {
-                _x ctrlSetFade 1;
-                _x ctrlCommit 0;
-            } forEach [
-                _statsPreviousPageCtrl,
-                _statsNextPageCtrl,
-                _statsCurrentPageCtrl
-            ];
+    // If there are no stats to show (unlikely), just hide everything
+    if (_statsTab isEqualTo []) exitWith {
+        call _fnc_hideEverything
+    };
+
```

Comment:
```suggestion

   GVAR(currentStatPage) = GVAR(currentStatPage) min floor ((count _statsTab) / 5); 
```
e.g. if you are on page 2 and go to a new item with only 3 stats total, it will jump back to a page with something on it

Reply:
This should really jump back to page 0 on item changes. Whoops.